staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -728,6 +728,43 @@ type CinderVolumeSource struct {
 	// More info: https://releases.k8s.io/HEAD/examples/mysql-cinder-pd/README.md
 	// +optional
 	ReadOnly bool `json:"readOnly,omitempty" protobuf:"varint,3,opt,name=readOnly"`
+	// Optional: points to a secret object that has enough information to connect
+	// to OpenStack. The secret data can contain fields from the "[Global]" section
+	// of the cloud configuration file. The values of these fields essentially
+	// override the values from the main configuration file set up by the
+	// administrator. This is helpful for example to customize the "domain-id",
+	// "tenant-id", "user-id" and "password" for specific pod(s)
+	// +optional
+	SecretRef *LocalObjectReference `json:"secretRef,omitempty" protobuf:"bytes,4,opt,name=secretRef"`
```

Comment:
keep the struct order the same between internal and external types (..., FSType, ReadOnly, SecretRef)

Reply:
Done